lib/global_config.js

```diff
@@ -256,4 +259,20 @@ exports.getKeepAlive = function () {
   return keepAlive;
 };
 
+let credentialManager = createDefaultCredentialManager();
+
+exports.setCustomCredentialManager = function (customCredentialManager) {
```

Comment:
it would be great to simulate users behaviour and create a test where we supply our own  custom implementation of credential manager. To keep it simple, perhaps we could do an in-memory implementation of it?

Reply:
In my opinion, the testing for this credential manager is the one I did not finish that you commented on. Those test cases are testing the valid credential manager and scenario of expected users input.

Reply:
I added testing for the custom credential manager. Please have a look at https://github.com/snowflakedb/snowflake-connector-nodejs/pull/663/files#diff-c92bed55d478910c2b3d0fa5930295b793b9d866ee42e7411ad6ac4fc15531adR12